portfolio/src/main/webapp/script.js

```diff
@@ -51,3 +73,27 @@ window.onclick = function(event) {
     }
   }
 }
+
+function plusSlides(n) {
+  showSlides(slideIndex += n);
+}
+
+function currentSlide(n) {
+  showSlides(slideIndex = n);
+}
+
+function showSlides(n) {
+  var i;
+  var slides = document.getElementsByClassName("mySlides");
+  var dots = document.getElementsByClassName("dot");
+  if (n > slides.length) {slideIndex = 1}    
```

Comment:
We should zero index the slides (so they go from zero to slides.length -1). zero-indexing is what will be expected by other engineers, and makes the array access on lines 97 and 98 simpler.

It would also let you use the mod operator to do the wrap around, if you wanted to. (optional)